web/src/components/__tests__/netflow-table.spec.tsx

```diff
@@ -49,4 +47,42 @@ describe('<NetflowTable />', () => {
     const expected = [FlowsSample[2], FlowsSample[0], FlowsSample[1]];
     expect(setFlows).toHaveBeenCalledWith(expected);
   });
+  it('should render a spinning slide and then the netflow rows', () => {
+    const wrapper = mount(<NetflowTable loading={true} flows={[]} setFlows={setFlows} columns={ColumnsSample} />);
+    expect(wrapper.find(NetflowTable)).toBeTruthy();
+    expect(wrapper.find('LoadingContents')).toHaveLength(1);
+    wrapper.setProps({
+      flows: FlowsSample.slice(0, FlowsSample.length),
+      loading: false
+    });
+    wrapper.update();
+    expect(wrapper.find('LoadingContents')).toHaveLength(0);
+    expect(wrapper.find('NoResultsFound')).toHaveLength(0);
+    expect(wrapper.find('ErrorState')).toHaveLength(0);
+    expect(wrapper.find(NetflowTableRow)).toHaveLength(FlowsSample.length);
+  });
+  it('should render a spinning slide and then a NoResultsFound message if no flows are found', () => {
+    const wrapper = mount(<NetflowTable loading={true} flows={[]} setFlows={setFlows} columns={ColumnsSample} />);
+    expect(wrapper.find(NetflowTable)).toBeTruthy();
+    expect(wrapper.find('LoadingContents')).toHaveLength(1);
```

Comment:
what is the difference here between calling find on a React.FC like `NetflowTableRow` and on a text like `'ErrorState'` ?

Reply:
'ErrorState' is not exported so it's not visible from the tests. Removing the quotes would not allow the tests to compile: `ReferenceError: ErrorState is not defined`

Reply:
Ho that's true. You created multiple `React.FC` in the same file.

For private usage, I would prefer [jsx const](https://github.com/netobserv/network-observability-console-plugin/blob/main/web/src/components/columns-modal.tsx#L102) in the same Function Component.